minvws/nl-kat-coordination
# Implement logging format configuration

### Changes

This PR implements a logging format option. It is implemented as an environment variable with the possible options "text" and "json" for each service (e.g. `OCTOPOES_LOGGING_FORMAT`, with the exception of Rocky, which just uses `LOGGING_FORMAT`)

This also replaces the Mula configuration in the same style (`MULA_LOGGING_JSON` is now replaced with `MULA_LOGGING_FORMAT`).

Also note that environment settings documentation is automatically generated.

This PR depends on #3175 (is also branches of `feature/structured-logging`)

### Issue link

Closes #3177 

### Demo

<img width="845" alt="image" src="https://github.com/minvws/nl-kat-coordination/assets/3471807/7075610a-49a8-4cbc-b5c1-3b4d4ca26624">

<img width="839" alt="image" src="https://github.com/minvws/nl-kat-coordination/assets/3471807/60de5f1d-004f-44f7-b878-321da96da9c0">

### QA notes

Toggle the logging format option between *text* and *json* to see difference.

---

### Code Checklist







---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.
